packages/lodestar/src/db/controller/impl/level.ts

```diff
@@ -79,19 +80,26 @@ export class LevelDbController extends EventEmitter implements IDatabaseControll
     await this.db.del(key);
   }
 
-  public search(opts: ISearchOptions): Promise<any> {
-    return new Promise<any[]>((resolve) => {
-      const searchData: any[] = [];
-      this.db.createValueStream({
-        gt: opts.gt,
-        lt: opts.lt,
-      }).on("data", function (data) {
-        searchData.push(data);
-      }).on("close", function () {
-        resolve(searchData);
-      }).on("end", function () {
-        resolve(searchData);
-      });
+  public async search(opts: ISearchOptions): Promise<any> {
+    const searchData: any[] = [];
+    for await (const data of this.searchStream(opts)) {
+      searchData.push(data);
+    }
+    return searchData;
+  }
+
+  public searchStream(opts: ISearchOptions): AsyncIterable<any> {
```

Comment:
does it work if you put unknow as a type. Isn't result always gonna be buffer|string?

Reply:
The whole DB controller uses `any` so it's hard to tell, I think this is something we need to improve as I forgot to deserialize and it compiled successfully.

Will create a separate issue to avoid any and work on that if you think so.

Reply:
Yeah any in controller is my bad, can't remember why I was doing that at the time

Reply:
Ok I created https://github.com/ChainSafe/lodestar/issues/685